### PR TITLE
[Docs] vllm docs

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Add vLLM docs following the doc structure

### CODE REVIEW

**Review by tosterberg — APPROVED** [review]

File: serving/docs/lmi_new/user_guides/vllm_user_guide.md [hunk 1]
```diff
@@ -1,9 +1,111 @@
-# vLLM Engine User Guide
+# vLLM Engine in User Guide
```
Comment (siddvenk) [inline serving/docs/lmi_new/user_guides/vllm_user_guide.md:1]: do you mean "in LMI"?

File: serving/docs/lmi_new/user_guides/vllm_user_guide.md [hunk 2]
```diff
@@ -1,9 +1,111 @@
...
-## Model Artifact Structure
+vLLM expect the model to be standard HuggingFace format.
+The source of the model could be:
 
-## Quick Start Configurations
+- model_id string from huggingface
+- s3 url that store the artifact that follows the HuggingFace model repo structure
+- A local path that contains everything in a folder follows HuggingFace model repo structure
+
+### Standard HuggingFace model artifacts
```
Comment (siddvenk) [inline serving/docs/lmi_new/user_guides/vllm_user_guide.md:12]: non blocking: we can update this to point to the single document covering model artifacts (deployment_guide/model_artifacts.md).

Can do in another PR

**Review by siddvenk — APPROVED** [review]
